packages/netlify-cms-core/index.d.ts

```diff
@@ -0,0 +1,255 @@
+declare module 'netlify-cms-core' {
+  import React, { ComponentType } from 'react';
+  import { Map } from 'immutable';
+
+  export type CmsBackendType
+    = 'git-gateway'
+    | 'github'
+    | 'gitlab'
+    | 'bitbucket'
+    | 'test-repo';
+
+  export type CmsMapWidgetType
+    = 'Point'
+    | 'LineString'
+    | 'Polygon';
+
+  export type CmsMarkdownWidgetButton
+    = 'bold'
+    | 'italic'
+    | 'code'
+    | 'link'
+    | 'heading-one'
+    | 'heading-two'
+    | 'heading-three'
+    | 'heading-four'
+    | 'heading-five'
+    | 'heading-six'
+    | 'quote'
+    | 'code-block'
+    | 'bulleted-list'
+    | 'numbered-list';
+
+  export type CmsFilesExtension
+    = 'yml'
+    | 'yaml'
+    | 'toml'
+    | 'json'
+    | 'md'
+    | 'markdown'
+    | 'html';
+
+  export type CmsCollectionFormatType
+    = 'yml'
+    | 'yaml'
+    | 'toml'
+    | 'json'
+    | 'frontmatter'
+    | 'yaml-frontmatter'
+    | 'toml-frontmatter'
+    | 'json-frontmatter';
+
+  export interface CmsField {
+    label: string;
+    name: string;
+    widget: string;
+    pattern?: string[];
+    default?: boolean | string | string[] | number | null;
+    format?: string;
+    dateFormat?: boolean | string;
+    timeFormat?: boolean | string;
+    media_library?: {
+      allow_multiple?: boolean;
+      config?: {
+        multiple?: boolean;
+      };
+    };
+    allow_add?: boolean;
+    field?: CmsField;
+    fields?: CmsField[];
+    decimals?: number;
+    type?: CmsMapWidgetType;
+    buttons?: CmsMarkdownWidgetButton[];
+    valueType?: number;
+    min?: number;
+    max?: number;
+    step?: number;
+    collection?: string;
+    displayFields?: string[];
+    valueField?: string;
+    multiple?: boolean;
+    optionsLength?: number;
+    options?: string[] | {
+      label: string;
+      value: string;
+    }[];
+  }
+
+  export interface CmsCollectionFile {
+    label: string;
+    name: string;
+    file: string;
+    fields: CmsField[];
+  }
+
+  export interface CmsCollection {
+    name: string;
+    fields: CmsField[];
+    label?: string;
+    label_singular?: string;
+    description?: string;
+    files?: CmsCollectionFile[];
+    folder?: string;
+    create?: boolean;
+    delete?: boolean;
+    filter?: {
+      field: string;
+      value: string;
+    };
+    identifier_field?: string;
+    extension?: CmsFilesExtension;
+    format?: CmsCollectionFormatType;
+    frontmatter_delimiter?: string[];
+    slug?: string;
+    preview_path?: string;
+    preview_path_date_field?: string;
+    editor?: {
+      preview: boolean;
+    };
+    summary?: string;
+  }
+
+  export interface CmsBackendParam {
+    name: CmsBackendType;
+    accept_roles?: string[];
+    repo?: string;
+    preview_context?: string;
+    auth_type?: string;
+    app_id?: string;
+    api_root?: string;
+    base_url?: string;
+    auth_endpoint?: string;
+  }
+
+  export interface CmsConfig {
+    backend?: CmsBackendType | CmsBackendParam;
```

Comment:
According to the schema here https://github.com/netlify/netlify-cms/blob/09ca1a7f0391b19b6d6065fcde93fc5a855a0f60/packages/netlify-cms-core/src/constants/configSchema.js#L32

backend should be an object

Reply:
Good remark, thanks! Didn't see the schema, to be honest... 😄  Let me fix this.